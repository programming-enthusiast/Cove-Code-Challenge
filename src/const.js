export const API_URL = "https://cove-coding-challenge-api.herokuapp.com/reservations"