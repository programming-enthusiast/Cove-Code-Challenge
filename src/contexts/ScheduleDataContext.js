import React from 'react'

export default React.createContext({
  scheduleData: [],
  setSchedulData: undefined
})