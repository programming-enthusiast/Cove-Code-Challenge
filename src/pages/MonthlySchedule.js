import './style.css'
import React, { useContext } from 'react'

import Calendar from '../components/calendar/Calendar'
import ScheduleDataContext from '../contexts/ScheduleDataContext'
import { getRoomName } from '../const'

export default function MonthlySchedulePage() {
  const { currentRoomId } = useContext(ScheduleDataContext)
  document.title = getRoomName(currentRoomId) + ' - Monthly Calendar'
  return (
    <div className="page-container">
      <h1>{document.title}</h1>
      <Calendar/>
    </div>
  )
}
