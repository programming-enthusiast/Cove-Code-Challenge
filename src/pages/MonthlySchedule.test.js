import React from "react"
import MonthlySchedule from "./MonthlySchedule"
import ScheduleDataContext from "../contexts/ScheduleDataContext"
import renderer from "react-test-renderer"
import { ROOM_LIST, SCHEDULE_DATA } from '../const'

describe("MonthlySchedule Component", () => {
  let renderedComponent

  let contextValue = { scheduleData: SCHEDULE_DATA, roomList: ROOM_LIST}

  beforeAll(() => {
    renderedComponent = renderer.create(
      <ScheduleDataContext.Provider value={contextValue}>
        <MonthlySchedule/>
      </ScheduleDataContext.Provider>
      )
  })

  it("should render correctly", () => {
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  })
})
