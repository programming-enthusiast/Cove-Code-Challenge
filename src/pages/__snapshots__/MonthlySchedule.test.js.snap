// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlySchedule Component should render correctly 1`] = `
<div
  className="page-container"
>
  <h1>
    undefined - Monthly Calendar
  </h1>
  <div
    className="calendar_container"
  >
    <div
      className="calendar__month_header_container"
    >
      <button
        className="calendar__header-button"
        disabled={false}
        onClick={[Function]}
      >
        &lt;
      </button>
      <div
        className="calendar__month-header__date"
      >
        April 2021
      </div>
      <button
        className="calendar__header-button"
        disabled={false}
        onClick={[Function]}
      >
        &gt;
      </button>
    </div>
    <div
      className="calendar_grid_container"
    >
      <div
        className="calendar-days-of-week"
      >
        <div
          className="calendar-day-of-week"
        >
          Sun
        </div>
        <div
          className="calendar-day-of-week"
        >
          Mon
        </div>
        <div
          className="calendar-day-of-week"
        >
          Tue
        </div>
        <div
          className="calendar-day-of-week"
        >
          Wed
        </div>
        <div
          className="calendar-day-of-week"
        >
          Thu
        </div>
        <div
          className="calendar-day-of-week"
        >
          Fri
        </div>
        <div
          className="calendar-day-of-week"
        >
          Sat
        </div>
      </div>
      <div
        className="calendar__week_container"
      >
        <div
          className="calendar__day_container sunday "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            28
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            29
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            30
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            31
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            1
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            2
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  saturday"
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            3
          </span>
          <div />
        </div>
      </div>
      <div
        className="calendar__week_container"
      >
        <div
          className="calendar__day_container sunday "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            4
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            5
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            6
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            7
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            8
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            9
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  saturday"
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            10
          </span>
          <div />
        </div>
      </div>
      <div
        className="calendar__week_container"
      >
        <div
          className="calendar__day_container sunday "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            11
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            12
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            13
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            14
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            15
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            16
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  saturday"
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            17
          </span>
          <div />
        </div>
      </div>
      <div
        className="calendar__week_container"
      >
        <div
          className="calendar__day_container sunday "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            18
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            19
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            20
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            21
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            22
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            23
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  saturday"
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            24
          </span>
          <div />
        </div>
      </div>
      <div
        className="calendar__week_container"
      >
        <div
          className="calendar__day_container sunday "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            25
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            26
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            27
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            28
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            29
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  "
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            30
          </span>
          <div />
        </div>
        <div
          className="calendar__day_container  saturday"
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "font": "bold",
                "fontSize": "20px",
              }
            }
          >
            1
          </span>
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
